client/dashboard/me/profile-personal-details/update-email/email-verification-banner.tsx

```diff
@@ -0,0 +1,169 @@
+import { resendEmailVerificationMutation } from '@automattic/api-queries';
+import { useMutation } from '@tanstack/react-query';
+import { Link } from '@tanstack/react-router';
+import { Button } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState, useEffect } from 'react';
+import Notice from '../../../components/notice';
+import type { UserSettings } from '@automattic/api-core';
+
+// Get email verification params from URL
+function getEmailVerificationParams() {
+	if ( typeof window === 'undefined' ) {
+		return { newEmailResult: null, verified: null };
+	}
+	const params = new URLSearchParams( window.location.search );
+	return {
+		newEmailResult: params.get( 'new_email_result' ),
+		verified: params.get( 'verified' ),
+	};
+}
+
+// Clean up email verification params from URL
+function cleanUpEmailVerificationParams() {
+	const params = new URLSearchParams( window.location.search );
+	params.delete( 'new_email_result' );
+	params.delete( 'verified' );
+	const newUrl = window.location.pathname + ( params.toString() ? '?' + params.toString() : '' );
+	window.history.replaceState( {}, '', newUrl );
+}
+
+interface EmailVerificationBannerProps {
+	userData: UserSettings;
+}
+
+export default function EmailVerificationBanner( { userData }: EmailVerificationBannerProps ) {
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const [ showResendButton, setShowResendButton ] = useState( true );
+	const [ showSuccessNotice, setShowSuccessNotice ] = useState( () => {
+		const { newEmailResult, verified } = getEmailVerificationParams();
```

Comment:
nitpick: I think we can extract this line outside of the clojure function and use those two fields everywhere they are need with a single function call.
Example
```suggestion
	const { newEmailResult, verified } = getEmailVerificationParams();
	const [ showSuccessNotice, setShowSuccessNotice ] = useState( () => {
```

Reply:
Extracted in [fe5690f](https://github.com/Automattic/wp-calypso/pull/106103/commits/fe5690fa907ae1c6f29e21417fd31a677bd287f3).